tools/cli/NearObjectCli.cxx

```diff
@@ -152,9 +148,17 @@ AddEnumOption(CLI::App* app, std::optional<EnumType>& assignTo)
 {
     std::string optionName{ std::string("--").append(magic_enum::enum_type_name<EnumType>()) };
 
-    return app->add_option(std::move(optionName), assignTo)
-        ->transform(CLI::CheckedTransformer(CreateEnumerationStringMap<EnumType>(), CLI::ignore_case))
-        ->capture_default_str();
+    const auto& map = CreateEnumerationStringMap<EnumType>();
+    std::ostringstream enumUsage;
+
+    auto it = std::cbegin(map);
+    enumUsage << "value in { " << it->second << "(" << it->first << ")";
+    for (it = std::next(std::cbegin(map)); it != std::cend(map); ++it) {
+        enumUsage << ", " << it->second << "(" << it->first << ")";
+    }
```

Comment:
Let's use a structured binding here for the map values to improve readability and comprehension. Someone coming upon this code won't know what `it->first` and `it->second` are, but a structured binding makes this super clear and sacrifices nothing:

```suggestion
    for (it = std::next(std::cbegin(map)); it != std::cend(map); ++it) {
        const auto& [name, value] = *it;
        enumUsage << ", " << value << "(" << name << ")";
    }
```

Reply:
On second thought, can we just use that directly for the loop?

```suggestion
    for (const auto& [name, value] : map) {
        enumUsage << ", " << value << "(" << name << ")";
    }
```

Reply:
I was using it directly for the loop at first, but then I ran into the fencepost problem with the comma separate list. I have made a change to match your original suggestion

Reply:
Why didn't the second suggestion (which is much cleaner) work?

Reply:
That's what I was referring to with the fencepost problem. If I do "for (const auto& [name, value] : map)", that will iterate through the entire map, meaning the output would start with a ", ". So to prevent that, I print the first value before the loop, then have the loop iterate through the map starting with the second entry so that the commas make sense

Reply:
Gotcha, I didn't read your comment and code carefully enough, thanks for the explanation. For future reference, a [`std::ranges::take_view`](https://en.cppreference.com/w/cpp/ranges/take_view) would also solve this problem.